pkg/controller/localvolumediscovery/localvolumediscovery_controller.go

```diff
@@ -202,6 +217,36 @@ func (r *ReconcileLocalVolumeDiscovery) updateDiscoveryStatus(instance *localv1a
 	return nil
 }
 
+func (r *ReconcileLocalVolumeDiscovery) deleteOrphanDiscoveryResults(instance *localv1alpha1.LocalVolumeDiscovery) error {
+	discoveryResultList := &localv1alpha1.LocalVolumeDiscoveryResultList{}
+	err := r.client.List(context.TODO(), discoveryResultList, client.InNamespace(instance.Namespace))
+	if err != nil {
+		return fmt.Errorf("failed to list LocalVolumeDiscoveryResult instances in namespace %q", instance.Namespace)
+	}
+
+	for _, discoveryResult := range discoveryResultList.Items {
+		nodeName := discoveryResult.Spec.NodeName
+		node := &corev1.Node{}
+		err = r.client.Get(context.TODO(), types.NamespacedName{Name: nodeName}, node)
+		if err != nil {
+			return fmt.Errorf("failed to get instance of node %q", nodeName)
+		}
+
+		matches := corev1helper.MatchNodeSelectorTerms(instance.Spec.NodeSelector.NodeSelectorTerms,
```

Comment:
this causes LSO to crash, if no nodeSelector was specified in CR..

Reply:
please ensure that we have a test case for this case.

Reply:
Thanks. Added a check to skip when no NodeSelector is specified [here](https://github.com/openshift/local-storage-operator/blob/9c72cb708080ed6693dc090f52ed3282ca3fb6cc/pkg/controller/localvolumediscovery/localvolumediscovery_controller.go#L227).  Added unit test for this case here [https://github.com/openshift/local-storage-operator/blob/9c72cb708080ed6693dc090f52ed3282ca3fb6cc/pkg/controller/localvolumediscovery/localvolumediscovery_controller_test.go#L275](url)